src/java.base/unix/classes/sun/nio/fs/UnixCopyFile.java

```diff
@@ -186,7 +186,7 @@ private static void copyDirectory(UnixPath source,
                 }
                 if (sfd >= 0) {
                     source.getFileSystem().copyNonPosixAttributes(sfd, dfd);
-                    close(sfd);
+                    close(sfd, x -> x.asIOException(source));
```

Comment:
This is the source fd so it's reading. It's not wrong to have it throw IOException, just a bit consistent to the other places where the exception is ignored.